src/SessionDurationMetric.php

```diff
@@ -0,0 +1,158 @@
+            'YTD' => $this->sessionDurationOneYear(),
+        ];
+
+        $data = Arr::get($lookups, $request->get('range'), ['result' => 0, 'previous' => 0]);
+
+        return (new ValueResult($data['result']))->previous($data['previous'])->format('00:00:00');
+    }
+
+    private function sessionDurationOneDay()
+    {
+        $analyticsData = app(Analytics::class)
+            ->performQuery(
+                Period::days(1),
+                'ga:avgSessionDuration',
+                [
+                    'metrics' => 'ga:avgSessionDuration',
+                    'dimensions' => 'ga:date',
+                ]
+            );
+
+        $results = collect($analyticsData->getRows());
+
+        return [
+            'result' => $results->last()[1] ?? 0,
+            'previous' => $results->first()[1] ?? 0,
+        ];
+    }
+
+    private function sessionDurationOneMonth()
+    {
+        $currentAnalyticsData = app(Analytics::class)->performQuery(
+            Period::create(Carbon::today()->startOfMonth(), Carbon::today()),
+            'ga:avgSessionDuration',
+            [
+                'metrics' => 'ga:avgSessionDuration',
+                'dimensions' => 'ga:yearMonth',
+            ]
+        );
+
+        $currentResults = collect($currentAnalyticsData->getRows());
+
+        $lastMonth = Carbon::today()->startOfMonth()->subMonths(1);
+        $previousAnalyticsData = app(Analytics::class)->performQuery(
+            Period::create($lastMonth->startOfMonth(), $lastMonth->endOfMonth()),
+            'ga:avgSessionDuration',
+            [
+                'metrics' => 'ga:avgSessionDuration',
+                'dimensions' => 'ga:yearMonth',
+            ]
+        );
+
+        $previousResults = collect($previousAnalyticsData->getRows());
+
+        return [
+            'previous' => $previousResults->last()[1] ?? 0,
+            'result' => $currentResults->last()[1] ?? 0,
+        ];
+    }
+
+    private function sessionDurationOneYear()
+    {
+        $currentAnalyticsData = app(Analytics::class)->performQuery(
+            Period::create(Carbon::today()->startOfYear(), Carbon::today()),
+            'ga:avgSessionDuration',
+            [
+                'metrics' => 'ga:avgSessionDuration',
+                'dimensions' => 'ga:year',
+            ]
+        );
+
+        $currentResults = collect($currentAnalyticsData->getRows());
+
+        $lastYear = Carbon::today()->startOfYear()->subYears(1);
+        $previousAnalyticsData = app(Analytics::class)->performQuery(
+            Period::create($lastYear->startOfYear(), $lastYear->endOfYear()),
+            'ga:avgSessionDuration',
+            [
+                'metrics' => 'ga:avgSessionDuration',
+                'dimensions' => 'ga:year',
+            ]
+        );
+
+        $previousResults = collect($previousAnalyticsData->getRows());
+
+        return [
+            'previous' => $previousResults->last()[1] ?? 0,
```

Comment:
Is this `??` available in our minimum PHP version? I assume so, but just wanted to make sure

Reply:
Yes, we've utilized it for our Page Views metric with no issues.